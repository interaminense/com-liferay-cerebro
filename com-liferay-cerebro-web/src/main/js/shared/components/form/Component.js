import autobind from 'autobind-decorator';
import Component, {Config} from 'metal-jsx';

import {checkRequired} from '../../util/form-validators';

class FormComponent extends Component {
	attached() {
		this.on('valueChanged', this.informChange);

		const {ancestorForm} = this.context;

		if (ancestorForm) {
			ancestorForm.addFormElement(this);
		}
	}

	detached() {
		const {ancestorForm} = this.context;

		if (ancestorForm) {
			ancestorForm.removeFormElement(this);
		}
	}

	getFormElementName() {
		return this.props.name;
	}

	getFormElementValue() {
		const {name, processValue} = this.props;

		let {value} = this.state;

		if (processValue) {
			value = processValue(value, name);
		}

		return value;
	}

	@autobind
	informChange() {
		if (this.context.ancestorForm) {
			this.context.ancestorForm.informChange(this.props.name);
		}
	}

	isRequired() {
		const {validator} = this.props;

		return validator && checkRequired(validator);
	}

	validateFormElement() {
		return true;
	}
}

FormComponent.PROPS = {
	name: Config.string().required(),
	processValue: Config.func()
};

FormComponent.STATE = {
	value: Config.any()
};

export {Config} from 'metal-jsx';

export default FormComponent;