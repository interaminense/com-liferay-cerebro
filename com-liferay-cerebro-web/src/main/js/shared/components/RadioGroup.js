import autobind from 'autobind-decorator';
import Component, {Config} from 'metal-jsx';
import getCN from 'classnames';

import {noop} from 'lodash';

class Option extends Component {
	@autobind
	handleClick() {
		const {onClick, value} = this.props;

		onClick(value);
	}

	render() {
		const {checked, disabled, displayInline, label, name} = this.props;

		const classes = getCN('custom-control', 'custom-radio', {
			['custom-control-inline']: displayInline
		});

		return (
			<div class={classes}>
				<label onClick={!disabled && this.handleClick}>
					<input
						{...this.otherProps()}
						checked={checked}
						class="custom-control-input"
						disabled={disabled}
						name={name}
						type="radio"
					/>

					<span class="custom-control-indicator" />

					<span class="custom-control-description">{label}</span>
				</label>
			</div>
		);
	}
}

Option.PROPS = {
	checked: Config.bool(),
	disabled: Config.bool().value(false),
	displayInline: Config.bool().value(false),
	label: Config.string(),
	name: Config.string(),
	onClick: Config.func(),
	value: Config.any()
};

class RadioGroup extends Component {
	render() {
		const {
			checked,
			children,
			disabled,
			inline,
			name,
			onChange
		} = this.props;

		const classes = getCN('form-check', {disabled});

		const displayInline = inline && children.length > 1;

		return (
			<div class={classes}>
				{children.map(child => {
					const {props} = child;

					child.props = {
						...props,
						checked: checked === props.value,
						disabled,
						displayInline,
						name,
						onClick: onChange
					};

					return child;
				})}
			</div>
		);
	}
}

RadioGroup.PROPS = {
	checked: Config.any(),
	disabled: Config.bool().value(false),
	inline: Config.bool().value(false),
	name: Config.string(),
	onChange: Config.func().value(noop)
};

RadioGroup.Option = Option;

export default RadioGroup;