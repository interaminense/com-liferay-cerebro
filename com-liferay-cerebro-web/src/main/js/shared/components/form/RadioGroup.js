import autobind from 'autobind-decorator';
import FormComponent, {Config} from './Component';
import RadioGroup from '../RadioGroup';

class FormRadioGroup extends FormComponent {
	created() {
		this.state.value = this.props.checked;
	}

	@autobind
	handleChange(newVal) {
		const {onChange} = this.props;

		this.state.value = newVal;

		if (onChange) {
			onChange(newVal);
		}
	}

	syncChecked(newVal) {
		this.state.value = newVal;
	}

	render() {
		const {checked, children, name} = this.props;

		return (
			<RadioGroup
				{...this.otherProps()}
				checked={checked}
				name={name}
				onChange={this.handleChange}
			>
				{children}
			</RadioGroup>
		);
	}
}

FormRadioGroup.PROPS = {
	checked: Config.any(),
	name: Config.string().required(),
	onChange: Config.func()
};

FormRadioGroup.STATE = {
	value: Config.bool()
};

FormRadioGroup.Option = RadioGroup.Option;

export default FormRadioGroup;