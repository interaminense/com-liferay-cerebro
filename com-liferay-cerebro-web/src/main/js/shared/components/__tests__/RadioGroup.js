import Component from 'metal-jsx';

import RadioGroup from '../RadioGroup';

describe('RadioGroup.Option', () => {
	let component;

	afterEach(() => {
		if (component) {
			component.dispose();
		}
	});

	it('should render Option', () => {
		component = new RadioGroup.Option();

		expect(component).toMatchSnapshot();
	});

	it('should render Option with value', () => {
		component = new RadioGroup.Option({value: 4});

		expect(component).toMatchSnapshot();
	});

	it('should render Option with label', () => {
		component = new RadioGroup.Option({label: 'foo bar'});

		expect(component).toMatchSnapshot();
	});

	it('should call `onClick` prop with value', () => {
		const onClick = jest.fn();
		const value = 'foo';

		component = new RadioGroup.Option({onClick, value});

		component.handleClick();

		expect(onClick).toHaveBeenCalledWith(value);
	});
});

class RadioGroupExample extends Component {
	render() {
		return (
			<RadioGroup {...this.otherProps()} name="foo">
				<RadioGroup.Option key={1} label="1" value={1} />
				<RadioGroup.Option key={2} label="2" value={2} />
				<RadioGroup.Option key={3} label="3" value={3} />
			</RadioGroup>
		);
	}
}

describe('RadioGroup', () => {
	let component;

	afterEach(() => {
		if (component) {
			component.dispose();
		}
	});

	it('should render RadioGroup', () => {
		component = new RadioGroup();

		expect(component).toMatchSnapshot();
	});

	it('should render RadioGroup with children', () => {
		component = new RadioGroupExample();

		expect(component).toMatchSnapshot();
	});

	it('should render RadioGroup with a selected value', () => {
		component = new RadioGroupExample({checked: 1});

		expect(component).toMatchSnapshot();
	});

	it('should render RadioGroup as disabled', () => {
		component = new RadioGroupExample({disabled: true});

		expect(component).toMatchSnapshot();
	});
});