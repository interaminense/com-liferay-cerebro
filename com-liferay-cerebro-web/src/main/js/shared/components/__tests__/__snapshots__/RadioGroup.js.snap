// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroup should render RadioGroup 1`] = `<div class="form-check"></div>`;

exports[`RadioGroup should render RadioGroup as disabled 1`] = `
<div class="form-check disabled">
  <div class="custom-control custom-radio">
    <label>
      <input disabled="" name="foo" class="custom-control-input" type="radio">
      <span class="custom-control-indicator"></span>
      <span class="custom-control-description">1</span>
    </label>
  </div>
  <div class="custom-control custom-radio">
    <label>
      <input disabled="" name="foo" class="custom-control-input" type="radio">
      <span class="custom-control-indicator"></span>
      <span class="custom-control-description">2</span>
    </label>
  </div>
  <div class="custom-control custom-radio">
    <label>
      <input disabled="" name="foo" class="custom-control-input" type="radio">
      <span class="custom-control-indicator"></span>
      <span class="custom-control-description">3</span>
    </label>
  </div>
</div>
`;

exports[`RadioGroup should render RadioGroup with a selected value 1`] = `
<div class="form-check">
  <div class="custom-control custom-radio">
    <label>
      <input checked="" name="foo" class="custom-control-input" type="radio">
      <span class="custom-control-indicator"></span>
      <span class="custom-control-description">1</span>
    </label>
  </div>
  <div class="custom-control custom-radio">
    <label>
      <input name="foo" class="custom-control-input" type="radio">
      <span class="custom-control-indicator"></span>
      <span class="custom-control-description">2</span>
    </label>
  </div>
  <div class="custom-control custom-radio">
    <label>
      <input name="foo" class="custom-control-input" type="radio">
      <span class="custom-control-indicator"></span>
      <span class="custom-control-description">3</span>
    </label>
  </div>
</div>
`;

exports[`RadioGroup should render RadioGroup with children 1`] = `
<div class="form-check">
  <div class="custom-control custom-radio">
    <label>
      <input name="foo" class="custom-control-input" type="radio">
      <span class="custom-control-indicator"></span>
      <span class="custom-control-description">1</span>
    </label>
  </div>
  <div class="custom-control custom-radio">
    <label>
      <input name="foo" class="custom-control-input" type="radio">
      <span class="custom-control-indicator"></span>
      <span class="custom-control-description">2</span>
    </label>
  </div>
  <div class="custom-control custom-radio">
    <label>
      <input name="foo" class="custom-control-input" type="radio">
      <span class="custom-control-indicator"></span>
      <span class="custom-control-description">3</span>
    </label>
  </div>
</div>
`;

exports[`RadioGroup.Option should render Option 1`] = `
<div class="custom-control custom-radio">
  <label>
    <input class="custom-control-input" type="radio">
    <span class="custom-control-indicator"></span>
    <span class="custom-control-description"></span>
  </label>
</div>
`;

exports[`RadioGroup.Option should render Option with label 1`] = `
<div class="custom-control custom-radio">
  <label>
    <input class="custom-control-input" type="radio">
    <span class="custom-control-indicator"></span>
    <span class="custom-control-description">foo bar</span>
  </label>
</div>
`;

exports[`RadioGroup.Option should render Option with value 1`] = `
<div class="custom-control custom-radio">
  <label>
    <input class="custom-control-input" type="radio">
    <span class="custom-control-indicator"></span>
    <span class="custom-control-description"></span>
  </label>
</div>
`;
